syntax = "proto3";

package rpc_package;

// define a service
service UserInfoService {
    // define the interface and data type
    rpc GetUserInfo (UserInfoRequest) returns (UserInfoReply) {}
}

// define the data type of request
message UserInfoRequest {
    string name = 1;
}

// define the data type of response
message UserInfoReply {
    int32 code = 5;
    string name = 1;
    int32 age = 2;
    int32 gender = 3;
    string date = 4;
    string message = 6;
}
